tensor_genn/algorithms/relu_ann.py

```diff
@@ -118,36 +150,47 @@ def convert(self, tf_model):
         cs_init = {"magnitude":10.0}
 
         # Fetch augmented weight matrices
-        g_weights, n_units = self.create_weight_matrices(tf_model)
-        gw_inds = [np.nonzero(gw) for gw in g_weights]
-        gw_vals = [g_weights[i][gw_inds[i]].reshape(-1) for i in range(len(g_weights))]
+        gw_inds, gw_vals, n_units, relevant_layers = self.create_weight_matrices(tf_model)
 
         # Define model and populations
         self.g_model = genn_model.GeNNModel("float","g_model")
         self.neuron_pops = []
         self.syn_pops = []
 
-        for i in range(1,n_layers):
+        for i in range(1,len(relevant_layers)+1):
```

Comment:
```python
for i, (inds, vals, n) in enumerate(zip(gw_inds, gw_vals, n_units)):
    pass
```
is probably a bit tidier and more pythonic. Also I don't think ``relevant_layers`` is necessary - you can determine the type of connectivity to use by just testing ``inds is None``